src/main/java/org/icgc/argo/program_service/converter/ToProtoProgramConverter.java

```diff
@@ -0,0 +1,137 @@
+import org.icgc.argo.program_service.PrimarySite;
+import org.icgc.argo.program_service.Program;
+import org.icgc.argo.program_service.model.entity.CancerEntity;
+import org.icgc.argo.program_service.model.entity.PrimarySiteEntity;
+import org.icgc.argo.program_service.model.entity.ProgramEntity;
+import org.icgc.argo.program_service.model.join.ProgramCancer;
+import org.icgc.argo.program_service.model.join.ProgramPrimarySite;
+import org.mapstruct.InheritConfiguration;
+import org.mapstruct.Mapper;
+import org.mapstruct.Mapping;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.UUID;
+
+@Mapper(config = ConverterConfig.class, uses = { CommonConverter.class })
+public interface ToProtoProgramConverter {
+
+  /**
+   * To Proto Converters
+   */
+  @Mapping(target = "clearField", ignore = true)
+  @Mapping(target = "clearOneof", ignore = true)
+  @Mapping(target = "mergeFrom", ignore = true)
+  @Mapping(target = "mergeShortName", ignore = true)
+  @Mapping(target = "mergeDescription", ignore = true)
+  @Mapping(target = "mergeName", ignore = true)
+  @Mapping(target = "mergeMembershipType", ignore = true)
+  @Mapping(target = "mergeCommitmentDonors", ignore = true)
+  @Mapping(target = "mergeSubmittedDonors", ignore = true)
+  @Mapping(target = "mergeGenomicDonors", ignore = true)
+  @Mapping(target = "mergeWebsite", ignore = true)
+  @Mapping(target = "mergeInstitutions", ignore = true)
+  @Mapping(target = "mergeCountries", ignore = true)
+  @Mapping(target = "mergeRegions", ignore = true)
+  @Mapping(target = "removeCancerTypes", ignore = true)
+  @Mapping(target = "removePrimarySites", ignore = true)
+  @Mapping(target = "mergeCreatedAt", ignore = true)
+  @Mapping(target = "mergeUpdatedAt", ignore = true)
+  @Mapping(target = "unknownFields", ignore = true)
+  @Mapping(target = "mergeUnknownFields", ignore = true)
+  @Mapping(target = "allFields", ignore = true)
+  @Mapping(target = "cancerTypesOrBuilderList", ignore = true)
+  @Mapping(target = "cancerTypesBuilderList", ignore = true)
+  @Mapping(target = "primarySitesOrBuilderList", ignore = true)
+  @Mapping(target = "primarySitesBuilderList", ignore = true)
+  @Mapping(source = "programCancers", target = "cancerTypesList")
+  @Mapping(source = "programPrimarySites", target = "primarySitesList")
+  Program programEntityToProgram(ProgramEntity entity);
+
+  @InheritConfiguration
+  List<Program> programEntitiesToPrograms(Collection<ProgramEntity> entities);
+
+	@Mapping(target = "mergeFrom", ignore = true)
+	@Mapping(target = "clearField", ignore = true)
+	@Mapping(target = "clearOneof", ignore = true)
+	@Mapping(target = "mergeId", ignore = true)
+	@Mapping(target = "mergeName", ignore = true)
+	@Mapping(target = "unknownFields", ignore = true)
+	@Mapping(target = "mergeUnknownFields", ignore = true)
+	@Mapping(target = "allFields", ignore = true)
+  Cancer cancerEntityToCancer(CancerEntity c);
+
+  @Mapping(target = "mergeFrom", ignore = true)
```

Comment:
Why not set unmappedTargetPolicy to ignore?

Reply:
defensive programming, because i want compilation to fail when a change is introduced and unaccounted for. This way, you have to explicitly ignore a target, as opposed to automatically. For instance, when i merged in master during the developement of this pr, a new field in ProgramEntity, called programEgoGroups was added. The mapstruct compilation failed, and made me ignore it, otherwise implement the correct logic to handle that conversion.